src/client/Streamer.ts

```diff
@@ -154,6 +154,26 @@ export class Streamer {
         });
     }
 
+    public async setStreamPreview(image: Buffer): Promise<void> {
+        if (!this.client.token)
+            throw new Error("Please login :)");
+        if (!this.voiceConnection?.streamConnection)
+            return;
+        const { streamKey } = this.voiceConnection.streamConnection;
+        const data = `data:image/jpeg;base64,${image.toString("base64")}`;
+
+        await fetch(`https://discord.com/api/v9/streams/${streamKey}/preview`, {
```

Comment:
Did you find that aiko's library `postPreview` or APIRequest wouldn't work here? I think it's a good idea to try to stick to it as much as possible to prevent getting possibly detected by Discord anti bot measures since the HTTP headers between requests might potentially be different (e.g. user-agent)

Reply:
Is it exported yet? I'll have a look later

Reply:
code snippet

```js
const voiceState = client.guilds.cache.get(server_id).members.me.voice;
await voiceState.postPreview('base64image');
// If error code is USER_NOT_STREAMING, try setting the property 'streaming' to true
// If this error occurs, it may be because the VOICE_STATE_UPDATE event is not cached properly
voiceState.streaming = true;
await voiceState.postPreview('base64image');
```

Reply:
Wouldn't it be simpler to just do

```typescript
await this.client.user.voice.postPreview(data);
```

since users can't join multiple voice channels anyway (bots can, as long as they're in separate servers)

Reply:
How about we expose the `APIRequest` thing for now? It's kinda weird for us to re-query information that we already have

Reply:
> Wouldn't it be simpler to just do
> 
> ```ts
> await this.client.user.voice.postPreview(data);
> ```
> 
> since users can't join multiple voice channels anyway (bots can, as long as they're in separate servers)

Users can also join multiple voice channels across different servers